sphinx/tutorial/cylc/furthertopics/message-triggers.rst

```diff
@@ -0,0 +1,429 @@
+.. _tutorial-cylc-message-triggers:
+
+Message Triggers
+================
+
+:term:`Message triggers <message trigger>` allow us to produce custom
+outputs for tasks before they have fully completed. These are also known as
+:term:`custom task outputs <message trigger>`.
+
+Explanation
+-----------
+
+We have seen :ref:`before <tutorial-qualifiers>` that tasks can have 
+:term:`qualifiers <qualifier>` for different 
+:term:`task states <task state>`.
+:term:`Message triggers <message trigger>` are essentially custom qualifiers.
+We can produce a bespoke output while our task is still running.
+This output could be, for example, a report or perhaps another task. 
+
+Usage
+-----
+
+:term:`Message triggers <message trigger>` are particularily useful if we have
+a long running task and we want to produce multiple tailored outputs whilst
+this task is running, rather than having to wait for the task to fully
+complete.
+
+We could also set up :term:`message triggers <message trigger>` to, for example,
+send an email to inform us that a submission has failed, making use of Cylc's 
+task event handling system. More information is available on these in the 
+`Cylc User Guide <https://cylc.github.io/doc/built-sphinx/task-implementation.html#task-messages`_.
+
+
+:term:`Message triggers <message trigger>` provide a superior solution to
+the problem of file system polling. We could, for example, design our suite
+such that we check if our task is finished by polling at intervals.
+It is inefficient to 'spam' task hosts with polling commands, it is preferable
+to set up a message trigger. 
+
+How to create a message trigger
+-------------------------------
+
+In order to get our suite to trigger messages, we need to:
+
+   * specify our custom message in a section called ``[[outputs]]`` within the
```

Comment:
This indentation should match the preceding paragraph (Sphinx interprets indented blocks as "block quotes").

```suggestion
* specify our custom message in a section called ``[[outputs]]`` within the
```